pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -453,12 +455,18 @@ private boolean isBrokerActive(String candidateBroker) throws KeeperException, I
      * @throws Exception
      */
     private String getLeastLoadedFromLoadManager(ServiceUnitId serviceUnit) throws Exception {
-        String lookupAddress = loadManager.get().getLeastLoaded(serviceUnit).getResourceId();
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("{} : redirecting to the least loaded broker, lookup address={}", pulsar.getWebServiceAddress(),
-                    lookupAddress);
+        ResourceUnit leastLoadedBroker = loadManager.get().getLeastLoaded(serviceUnit);
+        if (leastLoadedBroker != null) {
+            String lookupAddress = leastLoadedBroker.getResourceId();
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("{} : redirecting to the least loaded broker, lookup address={}",
+                        pulsar.getWebServiceAddress(), lookupAddress);
+            }
+            return lookupAddress;
+        } else {
+            LOG.warn("No broker is available for {}", serviceUnit);
+            throw new BrokerServiceException.ServiceUnitNotReadyException("No broker is available for " + serviceUnit);
```

Comment:
Ideally we should try to handle better these condition by using something like `Optional<>` instead of the exception.   For example, in this case we don't really need to get the stack-trace printed in logs. The warn message by itself would be enough.

Reply:
I understand that is outside the scope of this PR, just for future ref.